brian2/devices/device.py

```diff
@@ -494,23 +501,45 @@ def __setstate__(self, state):
         self.__dict__ = state
         self.__dict__["arrays"] = WeakKeyDictionary(self.__dict__["arrays"])
 
-    def get_array_name(self, var, access_data=True):
+    def get_array_name(self, var, access_data=True, get_pointer=False):
         # if no owner is set, this is a temporary object (e.g. the array
         # of indices when doing G.x[indices] = ...). The name is not
         # necessarily unique over several CodeObjects in this case.
         owner_name = getattr(var.owner, "name", "temporary")
 
         if isinstance(var, DynamicArrayVariable):
+            print("get_array_name", var.get_capsule())
             if access_data:
-                return f"_array_{owner_name}_{var.name}"
+                name = f"_array_{owner_name}_{var.name}"
             else:
-                return f"_dynamic_array_{owner_name}_{var.name}"
+                name = f"_dynamic_array_{owner_name}_{var.name}"
+
+            if get_pointer:
+                try:
+                    capsule = self.get_capsule(var)
+                    return (name, capsule)
+                except Exception as e:
+                    # If capsule creation fails, fall back to name only
+                    # This ensures backward compatibility even if something goes wrong
+                    import warnings
+
+                    warnings.warn(
+                        f"Could not create capsule for {var.name}: {e}. "
+                        f"Returning name only.",
+                        UserWarning,
+                        stacklevel=2,
+                    )
+                    return name
```

Comment:
I'm not sure if this is a good idea – if the code calls with `get_pointer == True` it will expect a tuple as return value, so it will fail with an exception a bit later anyway. For backwards compatibility, having the default argument `get_pointer=False` should be enough IMO.